lib/SymbolGraphGen/SymbolGraphASTWalker.cpp

```diff
@@ -159,25 +158,65 @@ bool SymbolGraphASTWalker::walkToDeclPre(Decl *D, CharSourceRange Range) {
       return false;
     }
 
+    // We only treat extensions to external types as extensions. Extensions to
+    // local types are directly associated with the extended nominal.
+    auto const shouldBeRecordedAsExtension =
+        Options.EmitExtensionBlockSymbols &&
+        !Extension->getModuleContext()->getNameStr().equals(
```

Comment:
This part of the expression is also used in SymbolGraph.cpp, it might be worth to refactor this into it's own function (maybe be a free function in the `symbolgraphgen` namespace).

Reply:
Resolved in c20c488e25f63a774c80292a95e0a67325dda154. I added the function to `SymbolGraphASTWalker` as that stores the `SymbolGraphOptions` which are required to evaluate the condition.